### DISCUSSION TITLE
How do I enable hardcore mode for NativeAOT?

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [dotnet/runtime#96934] [Q&A]
I want all code that is strictly not AOT compatible to fail my build at debug time, and point me to the list of offending assemblies. This way I can find and whine about not having NativeAOT support to the authors of those libraries.

**Comment by HighPerfDotNet** [comment]
Perhaps treating relevant  AOT warnings as errors would do the trick? A fair few external libraries that I use get those but they still work (at least what I need from them), might take a long time for authors to change old code, that ever happens.

**Comment by jkotas** [comment]
> AOT warnings as errors

https://github.com/dotnet/runtime/pull/96567
